helm/templates/_helpers.tpl

```diff
@@ -60,3 +60,29 @@ Create the name of the service account to use
 {{- default "default" .Values.serviceAccount.name }}
 {{- end }}
 {{- end }}
+
+{{/*
+Get the rabbitmq password secret.
+*/}}
+{{- define "idx.rabbitmq.secretPasswordName" -}}
+    {{- $rabbitmq := default dict .Values.rabbitmq -}}
+    {{- $auth := default dict $rabbitmq.auth -}}
+    {{- if $auth.existingPasswordSecret -}}
+        {{- printf "%s" (tpl .Values.rabbitmq.auth.existingPasswordSecret $) -}}
+    {{- else -}}
+        {{- printf "NEED_TO_SET_.Values.rabbitmq.auth.existingPasswordSecret_!!" -}}
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Get the solr password secret.
+*/}}
+{{- define "idx.solr.secretPasswordName" -}}
+    {{- $solr := default dict .Values.solr -}}
+    {{- $solrauth := default dict $solr.auth -}}
+    {{- if $solrauth.existingSecret -}}
+        {{- printf "%s" (tpl .Values.solr.auth.existingSecret $) -}}
+    {{- else -}}
+        {{- printf "NEED_TO_SET_.Values.solr.auth.existingPasswordSecret_!!" -}}
+    {{- end -}}
+{{- end -}}
```

Comment:
Why is this in the helpers.tpl? I'm not following the intent here. Seems complicated?

Reply:
yeah - good catch. I stole it from the [rmq bitnami chart](https://github.com/bitnami/charts/blob/main/bitnami/rabbitmq/templates/_helpers.tpl#L43) - they use it to substitute a default if not set - but that's not necessary here. Will excise